File: Python/Program/Weapon.py
from Equipment import Equipment

class Weapon(Equipment):
    def __init__(self, name: str, purpose: str, obtained_by: str, level: int, damage: int):
        super().__init__(name, purpose)
        self._obtained_by = obtained_by
        self._level = level
        self._damage = damage

    def get_obtained_by(self):
        return self._obtained_by

    def set_obtained_by(self, obtained_by: str):
        self._obtained_by = obtained_by

    def get_level(self):
        return self._level

    def set_level(self, level: int):
        self._level = level

    def get_damage(self):
        return self._damage

    def set_damage(self, damage: int):
        self._damage = damage

File: C++/Program/Weapon.cpp
#pragma once
#include <iostream>
#include <string>
#include "Equipment.cpp"

using namespace std;

class Weapon : public Equipment {
private:
    string obtainedBy;
    int level;
    int damage;

public:
    // Constructor
    Weapon(){}
    
    Weapon(string name, string purpose, string obtainedBy, int level, int damage)
        : Equipment(name, purpose) {
        this->obtainedBy = obtainedBy;
        this->level = level;
        this->damage = damage;
    }

    // Getter and Setter for obtainedBy
    string getObtainedBy() {
        return obtainedBy;
    }

    void setObtainedBy(string obtainedBy) {
        this->obtainedBy = obtainedBy;
    }

    // Getter and Setter for level
    int getLevel() {
        return level;
    }

    void setLevel(int level) {
        this->level = level;
    }

    // Getter and Setter for damage
    int getDamage() {
        return damage;
    }

    void setDamage(int damage) {
        this->damage = damage;
    }
};
